pkg/controller/argocd/util.go

```diff
@@ -793,16 +793,29 @@ func (r *ReconcileArgoCD) deleteClusterResources(cr *argoprojv1a1.ArgoCD) error
 }
 
 func (r *ReconcileArgoCD) removeManagedByLabelFromNamespace(namespace string) error {
-	ns := &corev1.Namespace{}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: namespace}, ns); err != nil {
+	nsList := &corev1.NamespaceList{}
+	listOption := client.MatchingLabels{
+		common.ArgoCDManagedByLabel: namespace,
+	}
+	if err := r.client.List(context.TODO(), nsList, listOption); err != nil {
 		return err
 	}
 
-	if ns.Labels == nil {
-		return nil
+	for _, n := range nsList.Items {
+		ns := &corev1.Namespace{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: n.Name}, ns); err != nil {
+			return err
+		}
+
+		if ns.Labels == nil {
+			return nil
+		}
+		delete(ns.Labels, common.ArgoCDManagedByLabel)
+		if err := r.client.Update(context.TODO(), ns); err != nil {
+			log.Error(err, fmt.Sprintf("failed to remove label from namespace [%s]", ns.Name))
+		}
```

Comment:
Do we want to test if that label still exists on the namespace, before running `delete()` and `Update()`?

Reply:
done!!